utils/litemap/src/map.rs

```diff
@@ -37,62 +35,56 @@ impl<K, V> LiteMap<K, V> {
     }
 }
 
-impl<K, V, S> LiteMap<K, V, S>
-where
-    S: Store<K, V>,
-{
-    /// Construct a new [`LiteMap`] with a given capacity
-    pub fn with_capacity(capacity: usize) -> Self {
+impl<K, V> LiteMap<K, V, Vec<(K, V)>> {
+    /// Convert a `Vec<(K, V)>` into a [`LiteMap`].
+    ///
+    /// # Safety
+    ///
+    /// The vec must be sorted and have no duplicate keys.
+    #[inline]
+    pub unsafe fn from_tuple_vec_unchecked(values: Vec<(K, V)>) -> Self {
         Self {
-            values: S::lm_with_capacity(capacity),
+            values,
             _key_type: PhantomData,
             _value_type: PhantomData,
         }
     }
 
-    /// The number of elements in the [`LiteMap`]
-    pub fn len(&self) -> usize {
-        self.values.lm_len()
-    }
-
-    /// Whether the [`LiteMap`] is empty
-    pub fn is_empty(&self) -> bool {
-        self.values.lm_is_empty()
-    }
-
-    /// Remove all elements from the [`LiteMap`]
-    pub fn clear(&mut self) {
-        self.values.lm_clear()
-    }
-
-    /// Reserve capacity for `additional` more elements to be inserted into
-    /// the [`LiteMap`] to avoid frequent reallocations.
-    ///
-    /// See [`Vec::reserve()`] for more information.
-    ///
-    /// [`Vec::reserve()`]: alloc::vec::Vec::reserve
-    pub fn reserve(&mut self, additional: usize) {
-        self.values.lm_reserve(additional)
+    /// Convert a [`LiteMap`] into a sorted `Vec<(K, V)>`.
+    #[inline]
+    pub fn into_tuple_vec(self) -> Vec<(K, V)> {
+        self.values
     }
+}
 
-    /// Convert a `Vec<(K, V)>` into a [`LiteMap`].
+impl<K: ?Sized, V: ?Sized> LiteMap<K, V, &'static[(&'static K, &'static V)]> {
```

Comment:
nit: must these be `'static`? I don't see any issues with letting them have lifetimes.

You can normalize them to the same lifetime since only the outer lifetime is actually real here.